moveit_core/robot_model/src/robot_model.cpp

```diff
@@ -1258,22 +1258,21 @@ void moveit::core::RobotModel::interpolate(const double* from, const double* to,
 void moveit::core::RobotModel::setKinematicsAllocators(const std::map<std::string, SolverAllocatorFn>& allocators)
 {
   // we first set all the "simple" allocators -- where a group has one IK solver
-  for (JointModelGroupMap::const_iterator it = joint_model_group_map_.begin(); it != joint_model_group_map_.end(); ++it)
+  for (JointModelGroup* jmg : joint_model_groups_)
   {
-    std::map<std::string, SolverAllocatorFn>::const_iterator jt = allocators.find(it->second->getName());
```

Comment:
Just to clarify.

`it->second->getName()` here is the same as `it->first`, is it?
Who wrote that code. -.-

This could be rewritten to avoid the map in other places in the file too (including the destructor)...

Reply:
Why do you want to get rid of the map? It's useful to have fast name-based access to the JMG.
However, here we need to iterate over all JMGs.

Reply:
You misunderstood my comment. I don't want to *get rid* of the map.
We don't want to find the correct names in a vector on each lookup.
However, there is more than one other place in this code, where we can iterate over the jmgs instead of the map...

Reply:
OK. I see. However, I don't want to search and fix for all these locations. I fix them as I encounter them ;-)

Reply:
That's a reasonable approach, yes...